src/benchmarklib/benchmark_runner.hpp

```diff
@@ -46,14 +46,15 @@ class BenchmarkRunner {
 
   // Calls _schedule_query if the scheduler is active, otherwise calls _execute_query and returns no tasks
   std::vector<std::shared_ptr<AbstractTask>> _schedule_or_execute_query(const QueryID query_id,
-                                                                        const std::function<void()>& done_callback);
+                                                                        const std::function<void()>& done_callback =
+                                                                            []() {});
 
   // Schedule and return all tasks for named_query
   std::vector<std::shared_ptr<AbstractTask>> _schedule_query(const QueryID query_id,
-                                                             const std::function<void()>& done_callback);
+                                                             const std::function<void()>& done_callback = []() {});
 
   // Execute named_query
-  void _execute_query(const QueryID query_id, const std::function<void()>& done_callback);
+  void _execute_query(const QueryID query_id, const std::function<void()>& done_callback = []() {});
```

Comment:
You added these defaults because of the `prepare` bool parameter you added, right? Since that is gone, I would also remove the default parameters for `done_callback` again.

Reply:
Mmh, I left this in on purpose, because I can't see a reason why we would want to enforce a callback.

Reply:
ok